### PR TITLE
[Topology] Rename Edge::pointBaryCoefs into Edge::getBarycentricCoordinates

### PR DESCRIPTION
**Pull request by epernod**
Following code review in PR #4053 regarding Triangle version, propagate the same method  name in Edge class

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Review by bakpaul — APPROVED** [review]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/Geometry/src/sofa/geometry/Edge.h [hunk 1]
```diff
@@ -90,18 +90,19 @@ struct Edge
...
     * @param	n0,n1: nodes of the edge
-    * @return	sofa::type::Vec<2, T> barycentric coefficients
+    * @return	sofa::type::Vec<2, T> barycentric coefficients of each vertex of the Edge.
     */
     template<typename Node,
         typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
         typename = std::enable_if_t<std::is_scalar_v<T>>
     >
-    static constexpr auto pointBaryCoefs(const Node& point, const Node& n0, const Node& n1)
+    static constexpr auto getBarycentricCoordinates(const Node& point, const Node& n0, const Node& n1)
```
Comment (fredroy) [inline Sofa/framework/Geometry/src/sofa/geometry/Edge.h:105]: For deprecation purpose (🤐) you might add an other static function with the former name calling this one, and with the `[[deprecated]]` flag

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/framework/Geometry/src/sofa/geometry/Edge.h (hunk 1 above)
Comment (epernod) [reply Sofa/framework/Geometry/src/sofa/geometry/Edge.h:105]: This method has been added 2 months ago https://github.com/sofa-framework/sofa/pull/4028
Do you think we need to add a deprecated process ?

Comment (bakpaul) [reply Sofa/framework/Geometry/src/sofa/geometry/Edge.h:105]: That is why I put the fast merge label, I also though that a deprecation mechanism wasn't necessary given the novelty of those methods.

Comment (fredroy) [reply Sofa/framework/Geometry/src/sofa/geometry/Edge.h:105]: I thought it was already integrated in the latest release, but it appears not so yeah we dont really care about the deprecation then

**Review by fredroy — APPROVED** [review]

**Pull request merged**